File: my_mindhelp_app/lib/models/counselor.dart
import 'package:json_annotation/json_annotation.dart';

part 'counselor.g.dart';

@JsonSerializable()
class Counselor {
  final String id;
  final String name;
  final String licenseNumber;
  final String gender;
  final String specialties;
  final String workLocation;
  final String workUnit;

  Counselor({
    required this.id,
    required this.name,
    required this.licenseNumber,
    required this.gender,
    required this.specialties,
    required this.workLocation,
    required this.workUnit,
  });

  factory Counselor.fromJson(Map<String, dynamic> json) => _$CounselorFromJson(json);
  Map<String, dynamic> toJson() => _$CounselorToJson(this);
}

@JsonSerializable()
class CounselorListResponse {
  final List<Counselor> counselors;
  final int total;
  final int page;
  final int pageSize;

  CounselorListResponse({
    required this.counselors,
    required this.total,
    required this.page,
    required this.pageSize,
  });

  factory CounselorListResponse.fromJson(Map<String, dynamic> json) => 
      _$CounselorListResponseFromJson(json);
  Map<String, dynamic> toJson() => _$CounselorListResponseToJson(this);
}

@JsonSerializable()
class CounselingCenter {
  final String id;
  final String name;
  final String address;
  final String phone;
  final String? website;
  final String? description;
  final bool hasOnlineService;

  CounselingCenter({
    required this.id,
    required this.name,
    required this.address,
    required this.phone,
    this.website,
    this.description,
    required this.hasOnlineService,
  });

  factory CounselingCenter.fromJson(Map<String, dynamic> json) => 
      _$CounselingCenterFromJson(json);
  Map<String, dynamic> toJson() => _$CounselingCenterToJson(this);
}

@JsonSerializable()
class CounselingCenterListResponse {
  final List<CounselingCenter> counselingCenters;
  final int total;
  final int page;
  final int pageSize;

  CounselingCenterListResponse({
    required this.counselingCenters,
    required this.total,
    required this.page,
    required this.pageSize,
  });

  factory CounselingCenterListResponse.fromJson(Map<String, dynamic> json) => 
      _$CounselingCenterListResponseFromJson(json);
  Map<String, dynamic> toJson() => _$CounselingCenterListResponseToJson(this);
}

@JsonSerializable()
class RecommendedDoctor {
  final String id;
  final String name;
  final String specialty;
  final String location;
  final String phone;
  final String? description;
  final int experienceCount;

  RecommendedDoctor({
    required this.id,
    required this.name,
    required this.specialty,
    required this.location,
    required this.phone,
    this.description,
    required this.experienceCount,
  });

  factory RecommendedDoctor.fromJson(Map<String, dynamic> json) => 
      _$RecommendedDoctorFromJson(json);
  Map<String, dynamic> toJson() => _$RecommendedDoctorToJson(this);
}

@JsonSerializable()
class RecommendedDoctorListResponse {
  final List<RecommendedDoctor> recommendedDoctors;
  final int total;
  final int page;
  final int pageSize;

  RecommendedDoctorListResponse({
    required this.recommendedDoctors,
    required this.total,
    required this.page,
    required this.pageSize,
  });

  factory RecommendedDoctorListResponse.fromJson(Map<String, dynamic> json) => 
      _$RecommendedDoctorListResponseFromJson(json);
  Map<String, dynamic> toJson() => _$RecommendedDoctorListResponseToJson(this);
}

File: backend/internal/dto/counselor.go
package dto

import "time"

// CounselorRequest 諮商師請求結構
type CounselorRequest struct {
	Name             string `json:"name" validate:"required"`
	LicenseNumber    string `json:"license_number" validate:"required"`
	Gender           string `json:"gender"`
	Specialties      string `json:"specialties"`
	LanguageSkills   string `json:"language_skills"`
	WorkLocation     string `json:"work_location"`
	WorkUnit         string `json:"work_unit"`
	InstitutionCode  string `json:"institution_code"`
	PsychologySchool string `json:"psychology_school"`
	TreatmentMethods string `json:"treatment_methods"`
}

// CounselorResponse 諮商師回應結構
type CounselorResponse struct {
	ID               string    `json:"id"`
	Name             string    `json:"name"`
	LicenseNumber    string    `json:"license_number"`
	Gender           string    `json:"gender"`
	Specialties      string    `json:"specialties"`
	LanguageSkills   string    `json:"language_skills"`
	WorkLocation     string    `json:"work_location"`
	WorkUnit         string    `json:"work_unit"`
	InstitutionCode  string    `json:"institution_code"`
	PsychologySchool string    `json:"psychology_school"`
	TreatmentMethods string    `json:"treatment_methods"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

// CounselorListResponse 諮商師列表回應結構
type CounselorListResponse struct {
	Counselors []CounselorResponse `json:"counselors"`
	Total      int64               `json:"total"`
	Page       int                 `json:"page"`
	PageSize   int                 `json:"page_size"`
}
